### ISSUE TITLE
test-laplace-petsc3d possibly broken?

### ISSUE THREAD

**Issue by ZedThree** [boutproject/BOUT-dev#2225]
The `data_circular_core` and `data_circular_core-sol` test cases have `mesh:paralleltransform = shiftedinterp`, but the option is now called `mesh:paralleltransform:type`, so they end up using the identity transform -- this passes. Making it actually use the shiftedinterp transform fails:

```
KSPConvergedReason is -9
terminate called after throwing an instance of 'BoutException'
  what():  Petsc3dAmg: inversion failed to converge.
```

which is `KSP_DIVERGED_NANORINF (residual norm became Not-a-number or Inf likely due to 0/0)`.

Possibly related is that the `data_slab_core` case doesn't seem to use one of the PETSc options:

```
WARNING! There are options you set that were not used!
WARNING! could be spelling mistake, etc!
There is one unused database option. It is:
Option left: name:-boutpetsclib_laplace_mg_levels_ksp_max_it value: 3
```

This is supposed to set the number of pre- and post-smoothing steps, and is required because at some point PETSc "fixed" this option and the default value now does one less step than before, changing the error convergence.

**Comment by johnomotani** [comment]
erk. That's disturbing, although it proves why errors for unused options is a good idea! Looks like I just messed up in #1803 - the tests that should have been using `shiftedinterp` were using `identity` when we merged that.

**Comment by ZedThree** [comment]
Possibly some more info, when using `laplace:pctype=gamg`, it fails with:

```
[0]PETSC ERROR: Petsc has generated inconsistent data
[0]PETSC ERROR: Computed maximum singular value as zero
```

It looks like the `mg_levels_ksp_max_it` is an option for the PETSc GAMG preconditioner, so not surprising it's not used by HYPRE.

@johnomotani Given that the test passes with the identity transform, is it sensible to just change it to use that?

**Comment by johnomotani** [comment]
> Given that the test passes with the identity transform, is it sensible to just change it to use that?

Not really, `shiftedinterp` vs. `parallel` is the most important difference between the 'circular' and 'slab' cases. The `shiftedinterp` case is the one we'll want in production, so it's the one we really want to test.

**Comment by johnomotani** [comment]
PS I think it's correct that the test passes with `identity` transform, because the Laplace solver and the forward operator both use `identity` so they are consistent. The problem is just that we really want to be testing `shiftedinterp`.

**Comment by ZedThree** [comment]
Ok, I've done the wrong thing in 0c04105 so that the test does something at least. I don't think I have the time or understanding to look into why the test is actually failing. It does sound like the matrix might be being constructed wrong? Bad boundary conditions, maybe?

**Comment by johnomotani** [comment]
> Ok, I've done the wrong thing in 0c04105 so that the test does something at least.

Sounds sensible, this isn't something that should hold up #2210, it definitely needs a separate PR. I'll try and have a look at some point, but it'll probably be a week or two at least.

**Comment by ZedThree** [comment]
Fixed in #2237
